iib/workers/tasks/opm_operations.py

```diff
@@ -415,6 +420,59 @@ def opm_generate_dockerfile(
     return dockerfile_path
 
 
+def insert_cache_into_dockerfile(dockerfile_path: str, local_cache_path: str) -> None:
+    """
+    Insert built cache into the Dockerfile.
+
+    :param str dockerfile_path: path to the generated Dockerfile.
+    :param str local_cache_path: path to the locally generated cache.
+    """
+    with open(dockerfile_path, 'r') as f:
+        file_data = f.read()
+
+    file_data = file_data.replace(
+        'RUN ["/bin/opm", "serve", "/configs", "--cache-dir=/tmp/cache", "--cache-only"]',
+        f'COPY {local_cache_path} /tmp/cache',
+    )
+
+    with open(dockerfile_path, 'w') as f:
+        f.write(file_data)
+
+
+def generate_cache_locally(base_dir: str, fbc_dir: str, local_cache_path: str) -> None:
+    """
+    Generate the cache for the index image locally before building it.
+
+    :param str base_dir: base directory where cache should be created.
+    :param str fbc_dir: directory containing file-based catalog (JSON or YAML files).
+    :param str local_cache_path: path to the locally generated cache.
+    :return: Returns path to generated cache
+    :rtype: str
+    :raises: IIBError when cache was not generated
+
+    """
+    from iib.workers.tasks.utils import run_cmd
+
+    cmd = [
+        'opm',
```

Comment:
Also I am not sure if it it still applies but I remember that @joelanford said in CLOUDDST-15194:
> modify the generated dockerfile such that the cache can be copied into the image rather than generated during the image build. We discussed that the version of opm used to generate the cache must match the version of opm in the binary image.

It means we should extract opm from the binary_image (or use opm based on from_index version)
In code we are using still the same opm version. I just wanted to point this out for discussion.

Reply:
Thanks for the review Jan! We did talk to Joe and it looks like the latest opm version 1.26.2 will work for building both 4.11 and 4.12 indexes. In earlier indexes, the new line is not introduced (verified locally). So this change along with OPM upgrade should be enough to unblock them